jurigged/runpy.py

```diff
@@ -249,11 +249,11 @@ def _get_main_module_details(error=ImportError):
 def _get_code_from_file(run_name, fname):
     # Check for a compiled file first
     decoded_path = os.path.abspath(os.fsdecode(fname))
-    with io.open_code(decoded_path) as f:
+    with io.open_code(decoded_path, encoding="utf8") as f:
```

Comment:
`open_code` is for binary files normally, I don't think it even takes an encoding argument. Most of the code in this file is also copied from `runpy` in the standard library, so it should already be well tested.